File: pset6/readability/readability.py
from cs50 import get_string


def main():
    para = get_string("Text: ")
    para_lower = para.lower()  # converting text to lower case for easy comparison

    grade = coleman_liau(para_lower)

    if(grade >= 16):
        print("Grade 16+")
    elif(grade < 1):
        print("Before Grade 1")
    else:
        print("Grade ", grade)


def coleman_liau(text):

    letter_count = 0
    word_count = 1  # word_count = 1 because last word doesn't have space
    sent_count = 0

    # creating a list of letters
    letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 
               'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

    for character in text:
        if character in letters:  # if a character is in letters list
            letter_count += 1
        elif character == " ":  # condition for word
            word_count += 1
        elif character == "." or character == "!" or character == "?" or character == " ":  # condition for sentence
            sent_count += 1

    L = (letter_count / word_count) * 100  # Average letters per 100 words
    S = (sent_count / word_count) * 100  # Average sentences per 100 words
    index = round(0.0588 * L - 0.296 * S - 15.8)  # Index calculation

    return index


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("TEXT: ");
    int text_length = strlen(text);
    int letter_count = 0;
    int word_count = 1; //the last word is not counted, thats why we started from 1
    int sent_count = 0;
    for (int i = 0; i < text_length; i++)
    {
        if (true) //always check this condition for counting letters
        {
            if ((text[i] >= 'a' || text[i] >= 'A') && (text[i] <= 'z' || text[i] <= 'Z')) //checking if the character is only alphabet or not
            {
                letter_count += 1;
            }
            else
            {
                letter_count += 0; //if letter is other than alphabet like(! . , % etc) then add nothing or 0
            }
        }
        if (true) //always check this condition for counting words
        {
            if (text[i] == ' ')
            {
                word_count += 1;
            }
        }
        if (true) //always check this condition for counting sentences
        {
            if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            {
                sent_count += 1;
            }
        }
    }
    float l = ((float)letter_count / (float)word_count) * 100.0;
    float s = ((float)sent_count / (float)word_count) * 100.0;
    int index = round(0.0588 * l - 0.296 * s - 15.8);
    if (index > 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
}